src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AzureEventHubsCmdletBase.cs

```diff
@@ -0,0 +1,164 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Eventhub;
+using Microsoft.Azure.Commands.EventHub.Models;
+using Microsoft.Azure.Management.EventHub.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Threading;
+using Microsoft.Azure.Commands.EventHub;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands
+{
+
+    public abstract class AzureEventHubsCmdletBase : AzureRMCmdlet
+    {
+        public const string InputFileParameterSetName = "InputFileParameterSet";
+        public const string SASRuleParameterSetName = "SASRuleParameterSet";
+        public const string EventHubParameterSetName = "EventHubParameterSet";
+        public const string ConsumerGroupParameterSetName = "ConsumerGroupParameterSet";
+        public const string RegenerateKeysSetName = "RegenerateKeysSet";
+                                                        
+        protected const string EventHubNamespaceVerb = "AzureRmEventHubNamespace";
+        protected const string EventHubNamespaceAuthorizationRulesVerb = "AzureRmEventHubNamespaceAuthorizationRules";
+        protected const string EventHubNamespaceKeysVerb = "AzureRmEventHubNamespaceKeys";
+        protected const string EventHubNamespaceRegenerateKeysVerb = "AzureRmEventHubNameSpaceRegenerateKeys";
```

Comment:
This is a verb and needs to be a singulr noun.  The cmdlet should be New-AzureRmEventHubNamespaceKey